dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorRISCV.java

```diff
@@ -403,132 +405,132 @@ public List<Event> visitAtomicXchg(AtomicXchg e) {
+			// 		https://elixir.bootlin.com/linux/v6.1/source/arch/riscv/include/asm/barrier.h#L72
+            // RISCV_FENCE(iorw,iorw) imposes ordering both on devices and memory
 			// 		https://github.com/westerndigitalcorporation/RISC-V-Linux/blob/master/linux/arch/riscv/include/asm/barrier.h
-			// Since the memory model says nothing about devices, we use RISCV_FENCE(rw,rw) which I think
-			// gives the ordering we want wrt. memory
+            // Since the memory model says nothing about devices, we use RISCV_FENCE(rw,rw) which I think
+            // gives the ordering we want wrt. memory
             case Tag.Linux.AFTER_SPINLOCK:
-				optionalMemoryBarrier = RISCV.newRWRWFence();
-				break;
-            // #define smp_mb__after_unlock_lock()	smp_mb()  /* Full ordering for lock. */
-            // 		https://elixir.bootlin.com/linux/v6.1/source/include/linux/rcupdate.h#L1008
+                optionalMemoryBarrier = RISCV.newRWRWFence();
+                break;
+			// #define smp_mb__after_unlock_lock()	smp_mb()  /* Full ordering for lock. */
+			// 		https://elixir.bootlin.com/linux/v6.1/source/include/linux/rcupdate.h#L1008
             // It seem to be only used for RCU related stuff in the kernel so it makes sense
             // it is defined in that header file
             case Tag.Linux.AFTER_UNLOCK_LOCK:
-				optionalMemoryBarrier = RISCV.newRWRWFence();
-				break;
-			default:
-				throw new UnsupportedOperationException("Compilation of fence " + e.getName() + " is not supported");
-		}
+                optionalMemoryBarrier = RISCV.newRWRWFence();
+                break;
+            default:
+                throw new UnsupportedOperationException("Compilation of fence " + e.getName() + " is not supported");
+        }
 
-		return eventSequence(
+        return eventSequence(
                 optionalMemoryBarrier
         );
-	}
+    }
 
-	public List<Event> visitRMWCmpXchg(RMWCmpXchg e) {
-		Register resultRegister = e.getResultRegister();
-		Expression address = e.getAddress();
-		Expression value = e.getMemValue();
-		String mo = e.getMo();
+    public List<Event> visitRMWCmpXchg(RMWCmpXchg e) {
+        Register resultRegister = e.getResultRegister();
+        Expression address = e.getAddress();
+        Expression value = e.getMemValue();
+        String mo = e.getMo();
 
-		Register dummy = e.getThread().newRegister(e.getResultRegister().getType());
-		Register statusReg = e.getThread().newRegister(e.getResultRegister().getType());
+        Register dummy = e.getThread().newRegister(e.getResultRegister().getType());
+        Register statusReg = e.getThread().newRegister(e.getResultRegister().getType());
         Label casEnd = newLabel("CAS_end");
         CondJump branchOnCasCmpResult = newJump(expressions.makeNEQ(dummy, e.getCmp()), casEnd);
-        
-        Load load = newRMWLoadExclusive(dummy, address, "");
+
+        Load load = newRMWLoadExclusive(dummy, address); // No mo on the load?
```

Comment:
Add TODO so IDE warning would just us this

Reply:
Will do